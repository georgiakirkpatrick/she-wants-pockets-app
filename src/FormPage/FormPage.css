.empty {
    height: 0;
}

.FormPage {
    padding: 5vh 10px 15vh;
    height: 90vh;
    overflow: scroll;
    position: relative
}

.FormPage form {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}

.FormPage h1 {
    text-align: center;
    padding: 0 0 25px;
    font-size: 30px;
    font-weight: 500;
}

.FormPage fieldset {
    padding: 10px;
    margin: 5px 0;
    background-color: rgb(238, 238, 238);
    border: none;
}