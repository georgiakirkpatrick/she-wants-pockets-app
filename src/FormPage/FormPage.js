import React from 'react'
import './FormPage.css'

const FormPage = props => {
    const title = props.title ? <h1>{props.title}</h1> : <div className='empty' />

    return (
        <section className='FormPage'>
            {title}
            <form>
                {props.children}
            </form>
        </section>
    )
}

export default FormPage;