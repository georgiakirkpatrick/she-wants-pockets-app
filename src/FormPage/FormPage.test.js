import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import FormPage from './FormPage'

describe('FormPage', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<FormPage />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(<FormPage />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})