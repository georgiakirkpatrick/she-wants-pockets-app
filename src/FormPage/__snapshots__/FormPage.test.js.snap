// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPage render the UI as expected 1`] = `
<section
  className="FormPage"
>
  <div
    className="empty"
  />
  <form
    id="form-page"
  >
    
  </form>
  <div />
</section>
`;
